hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/SqlPlanImpl.java

```diff
@@ -1044,7 +1042,13 @@ public boolean producesRows() {
         }
 
         @Override
-        public SqlResult execute(QueryId queryId, List<Object> arguments, long timeout, SqlSecurityContext ssc) {
+        public void checkPermissions(SqlSecurityContext context) {
```

Comment:
it would be more consistent if you invoke `extractPermissions` in `toExplainStatementPlan` in similar way as `toPlan`